module/userProjects/helper.js

```diff
@@ -2729,58 +2758,147 @@ module.exports = class UserProjectsHelper {
 			try {
-				// adding comments to check call back is called properly or not
-				console.log('<==================callback called====================>', transactionId, osid)
-				console.log('transactionId :', transactionId)
-				console.log('osid :', osid)
-				console.log('<==================callback called====================>')
-				// callback request structure nested so validating transactionId and osid here instead in validator.
-				if (transactionId == '' || osid == '') {
+				// Extract the content-disposition header to get the filename
+				const contentDisposition = requestObject.headers['content-disposition']
+				// Extract the Gotenberg trace header to get the transaction ID
+				const transactionId = requestObject.headers['gotenberg-trace']
+				// Match the filename from the content-disposition header
+				const match = contentDisposition && contentDisposition.match(/filename="(.+)"/)
+				// Set the filename or default to 'output.pdf' if not found
+				let filename = match && match[1] ? match[1] : 'output.pdf'
+				// Define the temporary folder path for the certificate
+				const certificateTempFolderPath = path.resolve(__dirname + '/../../certificateTempFolder')
+				// Define the file path to save the PDF
+				const filePath = certificateTempFolderPath + '/' + filename
+				// Create a writable stream to the file
+				const fileStream = fs.createWriteStream(filePath)
+
+				// Fetch project details based on the transaction ID
+				let projectDetails = await projectQueries.projectDocument({
+					'certificate.transactionId': transactionId,
+				})
+
+				// If no project details are found, throw an error
+				if (!projectDetails || !projectDetails.length > 0) {
 					throw {
-						status: HTTP_STATUS_CODE.bad_request.status,
-						message: CONSTANTS.apiResponses.TRANSACTION_ID_AND_OSID_REQUIRED,
+						message: CONSTANTS.apiResponses.PROJECT_NOT_FOUND,
 					}
 				}
-				let updateObject = {
-					$set: {},
-				}
 
-				// update osid and eligibility based on transactionId
-				updateObject['$set']['certificate.osid'] = osid
-				updateObject['$set']['certificate.message'] =
-					CONSTANTS.common.PROJECT_CERTIFICATE_GENERATED_SUCCESSFULLY
-				updateObject['$set']['certificate.issuedOn'] = new Date()
+				// Pipe the request stream to the file stream
+				requestObject.pipe(fileStream)
 
-				let projectDetails = await projectQueries.findOneAndUpdate(
-					{
-						'certificate.transactionId': transactionId,
-					},
-					updateObject
-				)
+				// Handle the end of the request stream
+				requestObject.on('end', async () => {
+					// Upload the PDF to the cloud and get the pre-signed URL
+					const pdfPreSignedUrl = await common_handler.uploadPdfToCloud(
+						filename,
+						projectDetails[0].userId,
+						certificateTempFolderPath
+					)
+					// Upload the SVG template to the cloud and get the pre-signed URL
+					const svgPreSignedUrl = await common_handler.uploadPdfToCloud(
```

Comment:
@Prajwal17Tunerlabs as we discussed we can update project with svg path before calling gotenberg, and remove it from here

Reply:
I have modified the code to add svgPath details before calling gotenberg as suggested.